app/src/main/java/org/mozilla/fenix/perf/ProfilerUtils.kt

```diff
@@ -0,0 +1,194 @@
+        val outputFile = createTemporaryFile(profileResult, context)
+        val response = networkCallToProfilerServer(outputFile, context)
+        val profileToken = decodeProfileToken(response)
+        outputFile.delete()
+        return PROFILER_DATA_URL + profileToken
+    }
+
+    private fun finishProfileSave(context: Context, url: String, onUrlFinish: (String) -> Unit) {
+        val clipboardManager = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
+        val clipData = ClipData.newPlainText("Profiler URL", url)
+        clipboardManager.setPrimaryClip(clipData)
+        onUrlFinish("URL saved to clipboard successfully. $url")
+    }
+
+    private fun saveProfileAsFile(profileSaveLauncher: ActivityResultLauncher<Intent>) {
+        val intent = Intent(Intent.ACTION_CREATE_DOCUMENT).apply {
+            addCategory(Intent.CATEGORY_OPENABLE)
+            type = "application/json"
+            putExtra(Intent.EXTRA_TITLE, "profiler.json")
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                putExtra(DocumentsContract.EXTRA_INITIAL_URI, "profiler.json")
+            }
+        }
+        profileSaveLauncher.launch(intent)
+    }
+
+    @Suppress("TooGenericExceptionCaught", "TooGenericExceptionThrown")
+    @Throws(Exception::class)
+    private fun createTemporaryFile(profileResult: ByteArray, context: Context): File {
+        GZIPInputStream(ByteArrayInputStream(profileResult)).use { gzipInputStream ->
+            val bufferedReader = BufferedReader(InputStreamReader(gzipInputStream))
+            val profileString = StringBuilder()
+            var line = bufferedReader.readLine()
+            while (line != null) {
+                profileString.append(line)
+                line = bufferedReader.readLine()
+            }
+            val outputDir = context.cacheDir
+            val outputFile = File.createTempFile("tempProfile", ".json", outputDir)
+            FileOutputStream(outputFile).use { fileOutputStream ->
+                fileOutputStream.write(profileResult)
+                return outputFile
+            }
+        }
+    }
+
+    @Throws(IOException::class)
+    private fun networkCallToProfilerServer(outputFile: File, context: Context): Response {
+        val request = Request(
+            url = PROFILER_API,
+            method = Request.Method.POST,
+            headers = MutableHeaders(
+                "Accept" to PROFILER_TOKEN
+            ),
+            body = Request.Body.fromFile(outputFile)
+        )
+        return context.components.core.client.fetch(request)
+    }
+
+    private fun decodeProfileToken(response: Response): String {
+        val jwtToken = StringBuilder("")
+        response.body.useStream { payload ->
+            payload.bufferedReader().use {
+                val jwt = it.readText()
+                val jwtSplit = jwt.split(".")
+                val decodedBytes = Base64.decode(jwtSplit[1], Base64.DEFAULT)
```

Comment:
Yeah we can. This is taken directly from this [payload script](https://github.com/firefox-devtools/profiler-server/blob/master/tools/decode_jwt_payload.py#L47) where they break the array into the 3 part they need with the split (instead of targetting with `[1]`)